File: dataset/output/CLAUDE_claude-3-5-sonnet-20240620/Task65_CLAUDE_claude-3-5-sonnet-20240620.java

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class FileReader {
    public static void main(String[] args) {
        if (args.length != 1) {
            System.out.println("Usage: java FileReader <filename>");
            return;
        }

        String filename = args[0];
        try {
            String content = readFile(filename);
            System.out.println("File content:");
            System.out.println(content);
        } catch (IOException e) {
            System.out.println("Error: " + e.getMessage());
        }
    }

    private static String readFile(String filename) throws IOException {
        StringBuilder content = new StringBuilder();
        try (BufferedReader reader = new BufferedReader(new FileReader(filename))) {
            String line;
            while ((line = reader.readLine()) != null) {
                content.append(line).append("\
");
            }
        }
        return content.toString();
    }
}

File: dataset/output/CLAUDE_claude-3-5-sonnet-20240620/Task65_CLAUDE_claude-3-5-sonnet-20240620.cpp

#include <iostream>
#include <fstream>
#include <string>
#include <filesystem>

void readFile(const std::string& filename) {
    if (filename.substr(filename.length() - 4) != ".txt") {
        std::cout << "Error: File must have a .txt extension" << std::endl;
        return;
    }

    if (!std::filesystem::exists(filename)) {
        std::cout << "Error: File '" << filename << "' does not exist" << std::endl;
        return;
    }

    std::ifstream file(filename);
    if (file.is_open()) {
        std::string line;
        std::cout << "File contents:" << std::endl;
        while (getline(file, line)) {
            std::cout << line << std::endl;
        }
        file.close();
    } else {
        std::cout << "Error: Unable to read file '" << filename << "'" << std::endl;
    }
}

int main(int argc, char* argv[]) {
    if (argc != 2) {
        std::cout << "Usage: " << argv[0] << " <filename>" << std::endl;
    } else {
        readFile(argv[1]);
    }
    return 0;
}
